facia-press/app/frontpress/FapiFrontPress.scala

```diff
@@ -340,43 +341,53 @@ trait FapiFrontPress extends EmailFrontPress with Logging {
   }
 
   private def enrichContent(collection: Collection, content: PressedContent, enriched: Option[EnrichedContent])(implicit executionContext: ExecutionContext): Response[EnrichedContent] = {
+    def extractCapiIdFromUrl(urlString: String): String = {
+      val uri = new URI(urlString)
+      uri.getPath()
+    }
 
     val beforeEnrichment = enriched.getOrElse(EnrichedContent.empty)
 
     val afterEnrichment = for {
       embedType <- content.properties.embedType if embedType == "json.html" || embedType == "interactive"
       embedUri <- content.properties.embedUri
     } yield {
-      val maybeUpdate = wsClient.url(embedUri).get().map { response =>
-        embedType match {
-          case "json.html" =>
+      val maybeUpdate = (embedType match {
+        case "json.html" =>
+          wsClient.url(embedUri).get().map { response =>
             Json.fromJson[EmbedJsonHtml](response.json) match {
               case JsSuccess(embed, _) =>
                 beforeEnrichment.copy(embedHtml = Some(embed.html))
               case _ =>
                 log.warn(s"An embed had invalid json format, and won't be pressed. ${content.properties.webTitle} for collection ${collection.id}")
                 beforeEnrichment
             }
-          case "interactive" =>
-            (response.json\"response"\"interactive"\"data"\"interactive").toOption.flatMap { j =>
-              Json.fromJson[EmbedInteractiveAtom](j) match {
-                case JsSuccess(embed, _) =>
-                  Some(beforeEnrichment.copy(embedHtml = Some(embed.html), embedCss = Some(embed.css), embedJs = Some(embed.mainJS)))
+          }
+        case "interactive" =>
+          val capiId = extractCapiIdFromUrl(embedUri)
+          capiClient.getResponse(ItemQuery(capiId)).map { response =>
+            response.interactive.flatMap { interactive =>
+              interactive.data match {
```

Comment:
Not much of an improvement, but for future interest `collect` provides a way to filter and map like you are doing here.